File: plugins/tene/test/fixtures/sample.ts
import { readFileSync } from 'node:fs'
import type { PaymentInput } from './types'
import React from 'react'
import * as utils from '../shared/utils'
const { legacy } = require('./legacy')

// function fakeInComment() {}
/* class FakeClass {} */
const s = 'function alsoFake() {}'
const re = /function notreal\(/g

export async function processPayment(input: PaymentInput): Promise<PaymentResult> {
  const result = await chargeCard(input)
  if (!result.ok) {
    return recordFailure(input, result)
  }
  return result
}

export const validateInput = (x: PaymentInput) => x.amount > 0

export class PaymentService {
  async charge(amount: number) {
    return this.gateway.send(amount)
  }
  private log(msg: string) { console.log(msg) }
}

export interface PaymentResult { ok: boolean }
export type Status = 'pending' | 'done'

const handlers = {}
handlers[key]()

File: plugins/tene/test/fixtures/Sample.java
package com.acme.payments;

import java.util.List;
import static java.util.Objects.requireNonNull;

// public class FakeInComment {}
public class PaymentService {
    private final Gateway gateway;

    public Result processPayment(Input input) throws PaymentException {
        Result r = chargeCard(input);
        if (!r.isOk()) {
            return recordFailure(input, r);
        }
        return r;
    }

    private void log(String msg) {
        System.out.println(msg);
    }
}

interface Gateway { }
